### PR TITLE
Update UntargetablePatch.cs Fixed targetPlayer and Nutcracker

### PR DESCRIPTION
**Pull request by Linkis20**
https://github.com/winstxnhdw/lc-hax/issues/241
Fixed targetPlayer and Nutcracker

Puffer:
Will see you and create smoke but will not bite you

Girl:
Will haunt you but will not move or be able to kill you

Mask:
Will look at you but can't kill you.

Giant:
Will not see you but can get killed if touched.

I tried to change PufferAI but got errors that playerIsInLOS did not exist
Idk why, PufferAI class has that property...

Also is line 10 !Setting.EnableGodMode needed?

### CODE REVIEW

**Comment by winstxnhdw** [comment]
`Setting.EnableGodMode` prevents infinite Masked spawn when player is grabbed during God mode.

**Comment by Linkis20** [comment]
Combined them

File: lc-hax/Scripts/Patches/UntargetablePatch.cs [hunk 1]
```diff
@@ -4,13 +4,35 @@
 using HarmonyLib;
 using Hax;
 
-[HarmonyPatch(typeof(EnemyAI), nameof(EnemyAI.PlayerIsTargetable))]
+[HarmonyPatch()]
```
Comment (winstxnhdw) [inline lc-hax/Scripts/Patches/UntargetablePatch.cs:7]: You don't need this line.

**Review by winstxnhdw — CHANGES_REQUESTED** [review]

**Review by winstxnhdw — APPROVED** [review]
Should be all good now.

File: lc-hax/Scripts/Patches/UntargetablePatch.cs (hunk 1 above)
Comment (winstxnhdw) [reply lc-hax/Scripts/Patches/UntargetablePatch.cs]: Okay, I am wrong here.

**Comment by xAstroBoy** [comment]
I would suggest to remove the !Setting.EnableGodMode and find a better approach to dodge the unlimited masked spawn if you remove demi-god

**Comment by winstxnhdw** [comment]
Why?

**Comment by xAstroBoy** [comment]
> Why?

Kinda ruins the game, makes it way too easy and people can easily figure that you are in godmode.

**Comment by winstxnhdw** [comment]
But you are literally in God mode..

**Comment by xAstroBoy** [comment]
> But you are literally in God mode..

Yeah but enemies should still target you regardless? Unless ignore is specified

**Comment by winstxnhdw** [comment]
Okay then should I just bring back `demigod` for this?

**Comment by xAstroBoy** [comment]
> Okay then should I just bring back `demigod` for this?

Or just remove the !setting.IsGodMode from the condition, i commented it out and even if a masked catches me i dont see any issues here.

**Comment by winstxnhdw** [comment]
Oh do you not get the infinite masked capture glitch?

**Comment by xAstroBoy** [comment]
> Oh do you not get the infinite masked capture glitch?

Yep, you can test it out, sure a masked will spawn but you can still dodge it out

**Comment by winstxnhdw** [comment]
@armorfid since it's no longer a problem in v49, I'll remove it.

**Pull request merged**